_codegen/main.go

```diff
@@ -298,6 +299,50 @@ func (f *testFunc) CommentWithoutT(receiver string) string {
 	return strings.Replace(f.Comment(), search, replace, -1)
 }
 
+func requireCommentParseIf(s string) string {
+	lines := strings.Split(s, "\n")
+	out := make([]string, 0, len(lines))
+	rePrefix := regexp.MustCompile(`//[ \t]*`)
+	ifDepth := 0
+	existsIfRequire := false
+	SomeFunctionLineIdx := 0
+
+	for i, line := range lines {
+		commentPrefix := rePrefix.FindString(line)
+		comment := strings.TrimSpace(line[2:])
+
+		if strings.HasSuffix(comment, "SomeFunction()") {
```

Comment:
I don't like the explicit reference to `SomeFunction()`. This should instead look for the first line of an example block by looking at indent. But maybe there is a reason why this wasn't enough?